sdks/python/apache_beam/ml/transforms/base.py

```diff
@@ -398,6 +398,17 @@ def save_attributes(
       artifact_location,
       **kwargs,
   ):
+    # if an artifact location is present, instead of overwriting the
+    # existing file, raise an error since the same artifact location
+    # can be used by multiple beam jobs and this could result in undesired
+    # behavior.
+    if FileSystems.exists(FileSystems.join(artifact_location,
+                                           _ATTRIBUTE_FILE_NAME)):
+      raise FileExistsError(
+          "The artifact location %s already exists and contains %s. Please "
+          "specify a different location." %
+          (artifact_location, _ATTRIBUTE_FILE_NAME))
```

Comment:
Good call - one possible future enhancement would be to support an overwrite argument that allows users to do this